### PR TITLE
Add DataCrunch.io backend

### PR DESCRIPTION
**Pull request by TheBits**
Hi!
I've added backend for datacrunch.io.

### CODE REVIEW

File: src/dstack/_internal/core/backends/datacrunch/api_client.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from tkinter import font
```
Comment (Egor-S) [inline src/dstack/_internal/core/backends/datacrunch/api_client.py]: This seems very odd

File: src/dstack/_internal/core/backends/datacrunch/compute.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            dockerized=False,
+            backend_data=None,
+        )
+
+        return launched_instance
+
+    def terminate_instance(
+        self, instance_id: str, region: str, backend_data: Optional[str] = None
+    ):
+        self.api_client.delete_instance(instance_id)
```
Comment (Egor-S) [inline src/dstack/_internal/core/backends/datacrunch/compute.py:151]: `compute.terminate_instance` must not raise an error, if an instance was not found. Is that the case there?

File: src/dstack/_internal/core/backends/datacrunch/compute.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            image=job.job_spec.image_name,
+        )
+
+        launched_instance = LaunchedInstanceInfo(
+            instance_id=instance.id,
+            ip_address=instance.ip,
+            region=instance.location,
+            ssh_port=22,
+            username="root",
+            dockerized=False,
```
Comment (Egor-S) [inline src/dstack/_internal/core/backends/datacrunch/compute.py]: Dockerized is True, if shim is used (i.e. the host is a VM)

File: src/dstack/_internal/core/backends/datacrunch/compute.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            description=name,
+            image=job.job_spec.image_name,
+        )
+
+        launched_instance = LaunchedInstanceInfo(
+            instance_id=instance.id,
+            ip_address=instance.ip,
+            region=instance.location,
+            ssh_port=22,
+            username="root",
```
Comment (Egor-S) [inline src/dstack/_internal/core/backends/datacrunch/compute.py:141]: Use "ubuntu" user if possible

File: src/dstack/_internal/core/models/backends/base.py [hunk 5]
```diff
@@ -9,6 +9,7 @@ class BackendType(str, enum.Enum):
     AZURE = "azure"
     DSTACK = "dstack"
     GCP = "gcp"
+    DATACRUNCH = "datacrunch"
```
Comment (Egor-S) [inline src/dstack/_internal/core/models/backends/base.py:12]: You also have to patch:

* `core/models/backends/__init__.py`
* `server/services/backends/__init__.py`
* `server/services/backends/configurators/datacrunch.py`
* `server/services/backends/config.py`

**Review by Egor-S — CHANGES_REQUESTED** [review]

File: src/dstack/_internal/core/backends/datacrunch/api_client.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import time
+from tracemalloc import start
```
Comment (Egor-S) [inline src/dstack/_internal/core/backends/datacrunch/api_client.py]: It seems unnecessary

**Review by Egor-S — CHANGES_REQUESTED** [review]

File: src/dstack/_internal/core/backends/datacrunch/api_client.py (hunk 6 above)
Comment (TheBits) [reply src/dstack/_internal/core/backends/datacrunch/api_client.py]: I've removed it

File: src/dstack/_internal/core/backends/datacrunch/compute.py (hunk 3 above)
Comment (TheBits) [reply src/dstack/_internal/core/backends/datacrunch/compute.py]: Ok! I've fixed it. I've understand how it works.

**Review by Egor-S — APPROVED** [review]

**Comment by Egor-S** [comment]
Please, upgrade `gpuhunt` version in `setup.py` to `0.0.1`
https://github.com/dstackai/gpuhunt/actions/runs/6954336643

**Pull request merged**